From b65dff31e68e0fe3fd13f3752ba176ab3e7e5309 Mon Sep 17 00:00:00 2001
From: Greg Sjaardema <gsjaardema@gmail.com>
Date: Thu, 13 Aug 2020 14:37:22 -0600
Subject: [PATCH] nvcc compiler should be EDG-based, but fails test

Our nvcc compilers (10.1.243 and 9.2.X) do not define the correct value for `FMT_USE_UDL_TEMPLATE` and then end up with an error later on in the build.  Explicitly search for `__NVCC__` symbol not being defined.   Might want to instead use `FMT_NVCC` or some other check, but the raw `__EDG_VERSION__` check is not working correctly for nvcc.
---
 include/fmt/format.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/spdlog/fmt/bundled/format.h b/include/spdlog/fmt/bundled/format.h
index dc88a9849..19a71fc05 100644
--- a/include/spdlog/fmt/bundled/format.h 2020-11-12 19:46:56.100687059 +0000
+++ b/include/spdlog/fmt/bundled/format.h 2020-11-12 19:48:08.307447631 +0000
@@ -144,7 +144,7 @@
 #  if FMT_USE_USER_DEFINED_LITERALS &&                         \
       (!defined(__EDG_VERSION__) || __EDG_VERSION__ >= 501) && \
       ((FMT_GCC_VERSION >= 604 && __cplusplus >= 201402L) ||   \
-       FMT_CLANG_VERSION >= 304)
+       FMT_CLANG_VERSION >= 304) && !defined(__NVCC__)
 #    define FMT_USE_UDL_TEMPLATE 1
 #  else
 #    define FMT_USE_UDL_TEMPLATE 0
